lib/iris/analysis/__init__.py

```diff
@@ -2348,13 +2348,31 @@ def regridder(self, src_grid, target_grid):
 
 class UnstructuredNearest(object):
     """
-    This is a nearest-neighbour interpolation and regridding scheme for
-    regridding cubes whose latitude and longitude coordinates are mapped to the
-    same dimensions, rather than being orthogonal on independent dimensions.
+    This is a nearest-neighbour regridding scheme for regridding data whose
+    horizontal (X- and Y-axis) coordinates are mapped to the *same* dimensions,
+    rather than being orthogonal on independent dimensions.
 
-    Currently only supports regridding, not interpolation.
+    For latitude-longitude coordinates, the nearest-neighbour distances are
+    computed on the sphere, otherwise flat Euclidean distances are used.
+
+    The source X and Y coordinates can have any shape.
+
+    The target grid must be of the "normal" kind, i.e. it has separate,
+    1-dimensional X and Y coordinates.
+
+    Source and target XY coordinates must have the same coordinate system,
+    which may also be None.
+    If any of the XY coordinates are latitudes or longitudes, then they *all*
+    must be.  Otherwise, the corresponding X and Y coordinates must have the
+    same units in the source and grid cubes.
+
+    .. Note::
+        Currently only supports regridding, not interpolation.
```

Comment:
This distinction isn't immediately obvious to me: interpolation is a numerical method which can support a regrid operation.
As this is the docstring for a class called `UnstructuredNearest` I suggest that this note is simply not required